File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/course/model/CourseDetailsResponseDTO.java
package kh.farrukh.sortmapperexamplejava.course.model;

import kh.farrukh.sortmapperexamplejava.teacher.model.TeacherResponseDTO;

public record CourseDetailsResponseDTO(
    Long id,
    String title,
    String description,
    String slogan,
    Long teacherId,
    TeacherResponseDTO teacher,
    Integer durationMonth,
    Double price
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/course/model/CourseDetailsResponseDTO.kt
package kh.farrukh.sortmapperexample.course.model

import kh.farrukh.sortmapperexample.teacher.model.TeacherResponseDTO

data class CourseDetailsResponseDTO(
    val id: Long,
    val title: String,
    val description: String,
    val slogan: String?,
    val teacherId: Long,
    val teacher: TeacherResponseDTO,
    val durationMonth: Int,
    val price: Double
)
